jerry-ext/include/jerryscript-ext/args.h

```diff
@@ -0,0 +1,132 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_ARGS_H
+#define JERRYX_ARGS_H
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include "jerryscript.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif /* __cplusplus */
+
+typedef struct jerryx_arg_t jerryx_arg_t;
+
+/**
+ * The iterator structor for JS arguments.
+ */
+typedef struct
+{
+  const jerry_value_t *js_arg_p; /**< the JS arguments */
+  const jerry_length_t js_arg_cnt; /**< the total num of JS arguments */
+  jerry_length_t js_arg_idx; /**< current index of JS argument */
+} jerryx_arg_js_iterator_t;
```

Comment:
We can force users to access the iterator via helpers only, that may not be a big problem, the already implemented transform functions did not need direct access to the fields either. But then we also need a helper function to create an interator instance, because jerryx_arg_transform_args and jerryx_arg_transform_args_with_this do need construction.